src/main/scala/no/ndla/learningpathapi/service/UpdateServiceComponent.scala

```diff
@@ -133,13 +133,27 @@ trait UpdateServiceComponent {
             searchIndexService.indexDocument(updatedLearningPath)
           } else if (existing.isPublished) {
             searchIndexService.deleteDocument(updatedLearningPath)
+            deleteIsBasedOnReference(learningPathId, updatedLearningPath)
           }
 
           Some(converterService.asApiLearningpath(updatedLearningPath, Option(owner)))
         }
       }
     }
 
+    private def deleteIsBasedOnReference(learningPathId: Long, updatedLearningPath: domain.LearningPath) = {
```

Comment:
2Den ligger inne som en optional, så for å slippe å matche på dette har jeg valgt å ta inn `learningPathId` isteden for å kjøre på med `updatedLearningPath.id.get` eller noe lignende. Jeg vet ikke hva som er best practice her?